### PR TITLE
Fix spelling

### PR DESCRIPTION
**Pull request by mdeweerd**
# Fix spelling

Fix multiple typos with codespell.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_db2_keywords.py [hunk 1]
```diff
@@ -376,7 +376,7 @@
     "STARTING",
     "STATEMENT",
     "STATIC",
-    "STATMENT",
+    "STATEMENT",
```
Comment (WittierDinosaur) [inline src/sqlfluff/dialects/dialect_db2_keywords.py:379]: this line needs deleting i think. Statement is two above so this is a dupe

File: src/sqlfluff/rules/capitalisation/CP02.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 from sqlfluff.core.rules import LintResult, RuleContext
 from sqlfluff.core.rules.crawlers import SegmentSeekerCrawler
 from sqlfluff.rules.capitalisation.CP01 import Rule_CP01
-from sqlfluff.utils.identifers import identifiers_policy_applicable
+from sqlfluff.utils.identifiers import identifiers_policy_applicable
```
Comment (WittierDinosaur) [inline src/sqlfluff/rules/capitalisation/CP02.py:8]: This is a breaking change without renaming the file

File: src/sqlfluff/rules/references/RF04.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 
 from sqlfluff.core.rules import BaseRule, LintResult, RuleContext
 from sqlfluff.core.rules.crawlers import SegmentSeekerCrawler
-from sqlfluff.utils.identifers import identifiers_policy_applicable
+from sqlfluff.utils.identifiers import identifiers_policy_applicable
```
Comment (WittierDinosaur) [inline src/sqlfluff/rules/references/RF04.py:9]: & here

File: src/sqlfluff/rules/references/RF05.py [hunk 4]
```diff
@@ -6,7 +6,7 @@
 
 from sqlfluff.core.rules import BaseRule, LintResult, RuleContext
 from sqlfluff.core.rules.crawlers import SegmentSeekerCrawler
-from sqlfluff.utils.identifers import identifiers_policy_applicable
+from sqlfluff.utils.identifiers import identifiers_policy_applicable
```
Comment (WittierDinosaur) [inline src/sqlfluff/rules/references/RF05.py:9]: & here

**Comment by mdeweerd** [comment]
I added the spelling exceptions to pyproject.toml in #5730, applied the fixes again, compared and updated this PR.

**Review by WittierDinosaur — APPROVED** [review]
LGTM

**Pull request merged**